### DISCUSSION TITLE
Can I build Malt on M1 Mac myself?

### DISCUSSION THREAD

**Discussion by Xayah-Hina** [bnpr/Malt#284] [Q&A] [answered]
Hi~

I'm new to blender and recently interested in NPR, and get attracted in Malt deeply.

I have noticed some info you provided about MacOS in [Malt 1.0 Preview](https://github.com/bnpr/Malt/discussions/231) and [doc](https://malt3d.com/FAQ/#will-malt-be-ported-to-vulkan), but I still wonder if I can build Malt on M1 Mac myself? I had some experience in PBR but don't quite know about blender plugins and python, and so would like to hear some advice from you~

**Comment by Xayah-Hina** [comment]
I found #217. Maybe MacOS would no longer be supported?

a little sad tho~

**Comment by pragma37** [comment, marked as answer]
I think the compilation on the Mac M1 should work as long as you have CMake and a C/C++ compiler installed.

That said, the built-in render pipeline won't work due to limits in the Mac OpenGL drivers.
So, unless you're interested in programming your own render pipeline there's not too much point in using Malt on Mac.

**Reply by Xayah-Hina** [reply]
I tried #217 and build by cmake myself, and got a "Python.h" not found error...

I export my python 3.10 include dir to the C_INCLUDE_PATH, but it seems didn't work.

I would make more effort to try more solution~ and by the way, I think maybe Vulkan backend would be a more cross-platform friendly choice?  I had some experience in shader programming and wonder if it would be hard for Malt to transfer OpenGL backend to Vulkan?

Melt is awesome, thank you~

**Reply by pragma37** [reply]
That `"Python.h" not found` error is likely caused by PyOpenGL_Accelerate. You can find an explanation on #279, but the short version is that PyOpenGL_Accelerate is not really needed for Malt to work.

>  I think maybe Vulkan backend would be a more cross-platform friendly choice?

Probably better, but still far from perfect on Mac. 
In any case that would take several months of work, I simply can't afford to put so much time into Malt at the moment.

> Melt is awesome, thank you~

Thanks! ^^
